src/js/gi/components/profile/Graph.jsx

```diff
@@ -0,0 +1,121 @@
+import React from 'react';
+
+class Graph extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.format = this.format.bind(this);
+  }
+
+  format(d) {
+    let val = d.value;
+    if (val === null || val === undefined) {
+      return 'No Data';
+    } else {
+      val = val.toFixed(this.props.decimals);
+      while (/(\d+)(\d{3})/.test(val.toString())) {
+        val = val.toString().replace(/(\d+)(\d{3})/, '$1' + ',' + '$2');
+      }
+      val = typeof d.percent !== 'undefined' ? '$' + val.replace('.0','') : val + '%';
+      return val;
+    }
+  }
+
+  render() {
+    const { width, height, padding, split } = this.props;
+    const viewBoxString = ['0', '0', width, height].join(' ');
+    const BarLabel = (props) => (
+      <text fontFamily="SourceSansPro-Regular, Source Sans Pro" fontSize="16" fill="#323A45" width="62">
+        <tspan x={props.x} y={props.y}>{props.text}</tspan>
+      </text>
+    );
+    const HollowUnderbar = ({top}) => <path d={`M80.544 28.8h128.912V3.2H80.544v25.6zM78 ${top || '0'}h134v32H78V0z`} fill="#F1F1F1" fillRule="nonzero"/>;
+    const SolidUnderbar = ({top}) => <path fill="#F1F1F1" d={`M78 ${top || 0}h134v32H78z`}/>;
+    const Underbar = ({top, value, children}) => {
+      return !!value ? <g><SolidUnderbar top={top}/>{children}</g> : <HollowUnderbar top={top}/>;
+    };
+    const ValueBar = ({top, color, percent}) => {
+      if (isNaN(percent)) { return null }
+      const width = 134 * (percent / this.props.max);
+      return <path fill={color} d={`M78 ${top}h${width}v32H78z`}/>;
+    };
+    const ValueLabel = ({y, text}) => (
+      <text fontFamily="SourceSansPro-Bold, Source Sans Pro" fontSize="16" fontWeight="bold" fill="#323A45">
+        <tspan x="220" y={y}>{text}</tspan>
+      </text>
+    );
+    const AverageMark = ({percent, text}) => {
```

Comment:
Other than `ValueBar`, these components don't directly use `Graph`'s props. They could probably be defined outside of the class, so that they don't get initialized on each render. Then for `ValueBar`, `this.props.max` could just be passed in as an arg.